sdk/ai/Azure.AI.Inference/src/Telemetry/OpenTelemetryScope.cs

```diff
@@ -0,0 +1,255 @@
+            if (endpoint.Port != 443)
+                _commonTags.Add(ServerPortKey, endpoint.Port);
+            // Set tags for reporting them to console.
+            foreach (KeyValuePair<string, object> kv in _commonTags)
+            {
+                _activity?.SetTag(kv.Key, kv.Value);
+            }
+            SetTagMaybe(GenAiRequestMaxTokensKey, requestOptions.MaxTokens);
+            SetTagMaybe(GenAiRequestTemperatureKey, requestOptions.Temperature);
+            if (requestOptions.AdditionalProperties != null && requestOptions.AdditionalProperties.ContainsKey("top_p"))
+            {
+                SetTagMaybe(
+                    GenAiRequestTopPKey,
+                    JsonSerializer.Deserialize<double?>(requestOptions.AdditionalProperties["top_p"]));
+            }
+            if (_traceContent)
+            {
+                // Log all messages as the events.
+                foreach (ChatRequestMessage message in requestOptions.Messages)
+                {
+                    TagList requestTags = new() {
+                    { GenAiSystemKey, GenAiSystemValue},
+                    { GenAiEventContent, GetContent(message)}
+                };
+                    _activity?.AddEvent(
+                        new ActivityEvent(
+                            $"gen_ai.{message.Role}.message",
+                            default,
+                            new ActivityTagsCollection(requestTags)
+                        )
+                    );
+                }
+            }
+            _duration = Stopwatch.StartNew();
+        }
+
+        public void RecordResponse(ChatCompletions response)
+        {
+            RecordResponseInternal(new SingleRecordedResponse(response, _traceContent));
+        }
+
+        public void UpdateStreamResponse<T>(T item)
+        {
+            if (!_enableTelemetry || item is not StreamingChatCompletionsUpdate)
+                return;
+            StreamingChatCompletionsUpdate castedItem = item as StreamingChatCompletionsUpdate;
+            m_recordedStreamingResponse.Update(castedItem);
+        }
+
+        public void RecordStreamingResponse()
+        {
+            RecordResponseInternal(m_recordedStreamingResponse);
+        }
+
+        /// <summary>
+        /// Log the error.
+        /// </summary>
+        /// <param name="e">Exception thrown by completion call.</param>
+        public void RecordError(Exception e)
+        {
+            if (!_enableTelemetry)
+                return;
+            if (!string.IsNullOrEmpty(_caller))
+                m_source.Write(_caller + ".Exception", e);
+            var errorType = e?.GetType()?.FullName;
+            var tags = _commonTags;
+            tags.Add(ErrorTypeKey, errorType);
+            s_duration.Record(_duration.Elapsed.TotalSeconds, tags);
+            if ( e is Azure.RequestFailedException requestFailed)
+            {
+                errorType = requestFailed.Status.ToString();
+            }
```

Comment:
it should happen before metric is recorded with `errorType` value

Reply:
Fixed.